File: 07 - DAT203.2x - Principles of Machine Learning/Labs/Module 1/CleanAdmissions.R
clean.admissions <- function(admissions){
  library(dplyr)
  admissions %>% mutate(admission_type_description =
                          ifelse(admission_type_description %in% 
                                   c('Not Available', 'NULL', 'Not Mapped'),
                                 'unknown', admission_type_description))
}

df <- maml.mapInputPort(1)
df <- clean.admissions(df)
maml.mapOutputPort('df')

File: 07 - DAT203.2x - Principles of Machine Learning/Labs/Module 1/CleanAdmissions.py
def prep_admissions(admissions):
    import pandas as pd
    admissions['admission_type_description'] = ['unknown' if ((x in ['Not Available', 'Not Mapped', 'NULL']) | (pd.isnull(x))) else x 
                                                 for x in admissions['admission_type_description']]
    return admissions

def azureml_main(df):
    df = prep_admissions(df)
    return df
